### ISSUE TITLE
Found security vulnerabilities in neuron-device-plugin:2.19.9.0 image

### ISSUE THREAD

**Issue by Bin-security** [aws-neuron/aws-neuron-sdk#817]
I am not sure this is the right place to report it. Some high-severity security vulnerabilities were found in the latest image neuron-device-plugin:2.19.9.0. Here is the command to reproduce it:
`trivy image --severity LOW,MEDIUM,HIGH,CRITICAL --ignore-unfixed --exit-code 3 --exit-on-eol 7 --scanners vuln public.ecr.aws/neuron/neuron-device-plugin:2.19.9.0`
The scanning results are here:
```
usr/bin/k8s-neuron-device-plugin (gobinary)

Total: 19 (HIGH: 19, CRITICAL: 0)

┌──────────────────────────┬─────────────────────┬──────────┬────────┬──────────────────────────────────────┬──────────────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│         Library          │    Vulnerability    │ Severity │ Status │          Installed Version           │              Fixed Version               │                            Title                             │
├──────────────────────────┼─────────────────────┼──────────┼────────┼──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/gogo/protobuf │ CVE-2021-3121       │ HIGH     │ fixed  │ v1.2.2-0.20190723190241-65acae22fc9d │ 1.3.2                                    │ gogo/protobuf: plugin/unmarshal/unmarshal.go lacks certain   │
│                          │                     │          │        │                                      │                                          │ index validation                                             │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2021-3121                    │
├──────────────────────────┼─────────────────────┤          │        ├──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/crypto      │ CVE-2020-29652      │          │        │ v0.0.0-20190820162420-60c769a6c586   │ 0.0.0-20201216223049-8b5274cf687f        │ crafted authentication request can lead to nil pointer       │
│                          │                     │          │        │                                      │                                          │ dereference                                                  │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2020-29652                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2020-7919       │          │        │                                      │ 0.0.0-20200124225646-8b5121be2f68        │ golang: Integer overflow on 32bit architectures via crafted  │
│                          │                     │          │        │                                      │                                          │ certificate allows for denial...                             │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2020-7919                    │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2020-9283       │          │        │                                      │ 0.0.0-20200220183623-bac4c82f6975        │ golang.org/x/crypto: Processing of crafted ssh-ed25519       │
│                          │                     │          │        │                                      │                                          │ public keys allows for panic                                 │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2020-9283                    │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2021-43565      │          │        │                                      │ 0.0.0-20211202192323-5770296d904e        │ empty plaintext packet causes panic                          │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2021-43565                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2022-27191      │          │        │                                      │ 0.0.0-20220314234659-1baeb1ce4c0b        │ golang: crash in a golang.org/x/crypto/ssh server            │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2022-27191                   │
├──────────────────────────┼─────────────────────┤          │        ├──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/net         │ CVE-2021-33194      │          │        │ v0.0.0-20191004110552-13f9640d40b9   │ 0.0.0-20210520170846-37e1c6afe023        │ infinite loop in ParseFragment                               │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2021-33194                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2022-27664      │          │        │                                      │ 0.0.0-20220906165146-f3363e06e74c        │ golang: net/http: handle server errors after sending GOAWAY  │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2022-27664                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2022-41723      │          │        │                                      │ 0.7.0                                    │ net/http, golang.org/x/net/http2: avoid quadratic complexity │
│                          │                     │          │        │                                      │                                          │ in HPACK decoding                                            │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2022-41723                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2023-39325      │          │        │                                      │ 0.17.0                                   │ golang: net/http, x/net/http2: rapid stream resets can cause │
│                          │                     │          │        │                                      │                                          │ excessive work (CVE-2023-44487)                              │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2023-39325                   │
├──────────────────────────┼─────────────────────┤          │        ├──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/text        │ CVE-2021-38561      │          │        │ v0.3.2                               │ 0.3.7                                    │ golang: out-of-bounds read in golang.org/x/text/language     │
│                          │                     │          │        │                                      │                                          │ leads to DoS                                                 │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2021-38561                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2022-32149      │          │        │                                      │ 0.3.8                                    │ golang: golang.org/x/text/language: ParseAcceptLanguage      │
│                          │                     │          │        │                                      │                                          │ takes a long time to parse complex tags                      │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2022-32149                   │
├──────────────────────────┼─────────────────────┤          │        ├──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ google.golang.org/grpc   │ GHSA-m425-mq94-257g │          │        │ v1.19.0                              │ 1.56.3, 1.57.1, 1.58.3                   │ gRPC-Go HTTP/2 Rapid Reset vulnerability                     │
│                          │                     │          │        │                                      │                                          │ https://github.com/advisories/GHSA-m425-mq94-257g            │
├──────────────────────────┼─────────────────────┤          │        ├──────────────────────────────────────┼──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ k8s.io/kubernetes        │ CVE-2019-11253      │          │        │ v1.14.3                              │ 1.13.12, 1.14.8, 1.15.5, 1.16.2          │ kubernetes: YAML parsing vulnerable to "Billion Laughs"      │
│                          │                     │          │        │                                      │                                          │ attack, allowing for remote denial...                        │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2019-11253                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2020-8558       │          │        │                                      │ 1.18.4, 1.17.7, 1.16.11                  │ kubernetes: node localhost services reachable via martian    │
│                          │                     │          │        │                                      │                                          │ packets                                                      │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2020-8558                    │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2021-25741      │          │        │                                      │ 1.19.15, 1.20.11, 1.21.5, 1.22.2         │ Symlink exchange can allow host filesystem access            │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2021-25741                   │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2023-3676       │          │        │                                      │ 1.28.1, 1.27.5, 1.26.8, 1.25.13, 1.24.17 │ kubernetes: Insufficient input sanitization on Windows nodes │
│                          │                     │          │        │                                      │                                          │ leads to privilege escalation                                │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2023-3676                    │
│                          ├─────────────────────┤          │        │                                      │                                          ├──────────────────────────────────────────────────────────────┤
│                          │ CVE-2023-3955       │          │        │                                      │                                          │ kubernetes: Insufficient input sanitization on Windows nodes │
│                          │                     │          │        │                                      │                                          │ leads to privilege escalation                                │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2023-3955                    │
│                          ├─────────────────────┤          │        │                                      ├──────────────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│                          │ CVE-2023-5528       │          │        │                                      │ 1.28.4, 1.27.8, 1.26.11, 1.25.16         │ kubernetes: Insufficient input sanitization in in-tree       │
│                          │                     │          │        │                                      │                                          │ storage plugin leads to privilege escalation...              │
│                          │                     │          │        │                                      │                                          │ https://avd.aquasec.com/nvd/cve-2023-5528                    │
└──────────────────────────┴─────────────────────┴──────────┴────────┴──────────────────────────────────────┴──────────────────────────────────────────┴──────────────────────────────────────────────────────────────┘

**Comment by ningziwen** [comment]
Thank you for your report. We are currently investigating and will follow up within 5 business days. If you become aware of any security concerns specific to AWS products and services, please do not hesitate to contact us at [aws-security@amazon.com](mailto:aws-security@amazon.com). Please refer to our [Vulnerability Reporting](https://aws.amazon.com/security/vulnerability-reporting/) page for more information.

**Comment by james-aws** [comment]
Thank you for the report. The fixes for all of the reported security vulnerabilities are available in the latest release 2.19.16.0 
 https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/containers/neuron-k8.html#neuron-k8-release-2-19-16-0

**Comment by Bin-security** [comment]
Thanks for fixing them quickly.
